db/migrate/20150710081302_create_cloud_ports_and_routers.rb

```diff
@@ -0,0 +1,52 @@
+class CreateCloudPortsAndRouters < ActiveRecord::Migration
+  def change
+    create_table :cloud_ports do |t|
+      t.string     :type
+      t.string     :name
+      t.string     :ems_ref
+      t.belongs_to :ems,               :type => :bigint
+      t.belongs_to :cloud_network,     :type => :bigint
+      t.belongs_to :cloud_subnet,      :type => :bigint
+      t.string     :mac_address
+      t.string     :status
+      t.boolean    :admin_state_up
+      t.string     :device_owner
+      t.string     :device_ref
+      t.integer    :device_id,         :type => :bigint
+      t.string     :device_type
```

Comment:
I've identified at least two devices in the basic examples, nova instance and router. But according to doc, there can be more.